app/src/main/resources/static/js/calender.js

```diff
@@ -14,6 +14,17 @@ document.addEventListener('DOMContentLoaded', function() {
         }
     }
 
+    for(let i = 0; i < datepickerRetroativo.length; i++) {
+        if (document.getElementById("formCheckin") != null) {
```

Comment:
Eu acho que esse `if` deveria ficar fora do for. Estamos verificando se o `formCheckin` existe a cada iteração. Pelo que entendi,  nem deveríamos executar esse `for` caso `formCheckin` não exista, então, deveríamos ter o seguinte:

```suggestion
        if (document.getElementById("formCheckin") != null) {
            for(let i = 0; i < datepickerRetroativo.length; i++) {
              // ... 
            }
        }
```